src/sql/src/partition.rs

```diff
@@ -38,117 +40,105 @@ macro_rules! between_string {
+                    "{:02x}",
+                    start
+                )))),
+            });
+        } else {
+            // Create the middle rules, for example: trace_id >= '10' AND trace_id < '20'.
+            rules.push(between_string!(
+                ident_expr,
+                format!("{:02x}", start),
+                format!("{:02x}", end)
+            ));
+        }
 
-    Partitions {
-        column_list: vec![ident],
-        exprs: rules,
+        current_boundary = end;
     }
+
+    rules
 }
 
 #[cfg(test)]
 mod tests {
+    use std::collections::HashMap;
+
     use sqlparser::ast::Expr;
     use sqlparser::dialect::GenericDialect;
     use sqlparser::parser::Parser;
+    use uuid::Uuid;
 
     use super::*;
 
+    #[test]
+    fn test_partition_rules_for_uuid() {
+        assert!(check_distribution(16, 10000));
+        assert!(check_distribution(32, 10000));
+        assert!(check_distribution(64, 10000));
+        assert!(check_distribution(128, 10000));
+        assert!(check_distribution(256, 10000));
+    }
+
     #[test]
     fn test_rules() {
         let expr = vec![
-            "trace_id < '1'",
-            "trace_id >= '1' AND trace_id < '2'",
-            "trace_id >= '2' AND trace_id < '3'",
-            "trace_id >= '3' AND trace_id < '4'",
-            "trace_id >= '4' AND trace_id < '5'",
-            "trace_id >= '5' AND trace_id < '6'",
-            "trace_id >= '6' AND trace_id < '7'",
-            "trace_id >= '7' AND trace_id < '8'",
-            "trace_id >= '8' AND trace_id < '9'",
-            "trace_id >= '9' AND trace_id < 'A'",
-            "trace_id >= 'A' AND trace_id < 'B' OR trace_id >= 'a' AND trace_id < 'b'",
-            "trace_id >= 'B' AND trace_id < 'C' OR trace_id >= 'b' AND trace_id < 'c'",
-            "trace_id >= 'C' AND trace_id < 'D' OR trace_id >= 'c' AND trace_id < 'd'",
-            "trace_id >= 'D' AND trace_id < 'E' OR trace_id >= 'd' AND trace_id < 'e'",
-            "trace_id >= 'E' AND trace_id < 'F' OR trace_id >= 'e' AND trace_id < 'f'",
-            "trace_id >= 'F' AND trace_id < 'a' OR trace_id >= 'f'",
+            "trace_id < '10'",
+            "trace_id >= '10' AND trace_id < '20'",
+            "trace_id >= '20' AND trace_id < '30'",
+            "trace_id >= '30' AND trace_id < '40'",
+            "trace_id >= '40' AND trace_id < '50'",
+            "trace_id >= '50' AND trace_id < '60'",
+            "trace_id >= '60' AND trace_id < '70'",
+            "trace_id >= '70' AND trace_id < '80'",
+            "trace_id >= '80' AND trace_id < '90'",
+            "trace_id >= '90' AND trace_id < 'a0'",
+            "trace_id >= 'a0' AND trace_id < 'b0'",
+            "trace_id >= 'b0' AND trace_id < 'c0'",
+            "trace_id >= 'c0' AND trace_id < 'd0'",
+            "trace_id >= 'd0' AND trace_id < 'e0'",
+            "trace_id >= 'e0' AND trace_id < 'f0'",
+            "trace_id >= 'f0'",
```

Comment:
we need to handle big letter and small letter together like previous rule

Reply:
Is it necessary to handle partition rules? It makes things too complicated. Most UUIDs are lowercase. Can we change all the UUIDs to lowercase when applying the rules? Actually, in the OLTP trace scenario, we will use https://github.com/GreptimeTeam/greptimedb/blob/main/src/servers/src/otlp/utils.rs#L21-L23 to convert the random bytes to lower-hex strings.

Reply:
@sunng87 @waynexia I found out the partition rules will be complicated(typically, it's hard to program) and hard to read with an increase in partition number:

In opentelemetry trace spec:

1. TraceId will be a 16 random bytes;
2. When retrieving the trace id, it will return **lowercase** hex-encoded format;

Since we always handle lowercase hex, why wouldn't we simplify it?

Reply:
https://opentelemetry.io/docs/specs/otel/trace/api/#retrieving-the-traceid-and-spanid

It makes to me. We can support lowercase only.